infra/cmake/packages/BoostConfig.cmake

```diff
@@ -60,7 +60,7 @@ endfunction(_Boost_Build)
 if (NOT BUILD_BOOST)
   # BoostConfig.cmake does not honor QUIET argument at least till cmake 1.70.0.
   # Thus, don't try to find_package if you're not entirely sure you have boost.
-  find_package(Boost 1.58.0 QUIET COMPONENTS log program_options filesystem system)
+  find_package(Boost 1.84.0 QUIET COMPONENTS log program_options filesystem system)
```

Comment:
Boost 1.58.0 does not support C++17. 
Since `auto_ptr`is deprecated in c++17, if we compile with boost old version, it throws an error message like `auto_ptr` is not found in `std` namespace. 

https://www.boost.org/users/history/version_1_58_0.html#:~:text=Boost%27s%20primary%20test,4.4.7%2C%204.8.4%2C%204.9.2 

> Boost's primary test compilers are:
> Linux:
> Clang: 3.4
> Clang, C++14: 3.5, 3.6
> GCC: 4.4.7, 4.9.2
> GCC, C++98: 4.8.1, 4.8.2
> GCC, C++11: 4.4.7, 4.8.4, 4.


----------------------

So, Let's upgrade Boost version.

Reply:
Just for convey information, `auto_ptr` is removed in c++17.